### PR TITLE
Table writer 4: Refactor table writer API usage

### PR DESCRIPTION
**Pull request by gggrace14**
Deprecate the notion of write protocol, to adapt to the change
in Velox. Clean up enum values that are not being used.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by gggrace14** [comment]
> Looks like the commit is mixed up with an earlier commit from rebecca ? I see some code changes from join optimization ?

Yes, thanks for reminding, Nikhil! Not sure when local branch was messed up. Just uploaded the correct commit :)

**Comment by ajaygeorge** [comment]
@gggrace14 Curious about the commit title/message. What does `
Table writer 4` mean?.  Is it v4 or 4th commit in a series of commits?.
Can you please update the wording to be a bit more clear.

**Comment by ajaygeorge** [comment]
@gggrace14 Also there is a file conflict with velox. You need to rebase with master.

File: presto-native-execution/presto_cpp/presto_protocol/WriteProtocol.cpp [hunk 1]
```diff
@@ -35,25 +35,27 @@ std::string makePartitionUpdate(
...
-     ("rowCount", numWrittenRows)
-     // TODO(gaoge): track and send the fields when inMemoryDataSizeInBytes, onDiskDataSizeInBytes
-     // and containsNumberedFileNames are needed at coordinator when file_renaming_enabled are turned on,
-     // https://fburl.com/18vim7k4.
-     ("inMemoryDataSizeInBytes", 0)
-     ("onDiskDataSizeInBytes", 0)
-     ("containsNumberedFileNames", true));
+ auto partitionUpdateJson = folly::toJson(
+     folly::dynamic::object
+         ("name", hiveWriterParameters->partitionName().value_or(""))
```
Comment (gggrace14) [inline presto-native-execution/presto_cpp/presto_protocol/WriteProtocol.cpp:40]: Doing some formatting, but only the line of the "name" field changes, for this function

**Comment by gggrace14** [comment]
> @gggrace14 Curious about the commit title/message. What does ` Table writer 4` mean?. Is it v4 or 4th commit in a series of commits?. Can you please update the wording to be a bit more clear.

Hi 

> @gggrace14 Also there is a file conflict with velox. You need to rebase with master.

Hi @ajaygeorge thank you for feedback! As this depends on velox submodule, we can only get all tests green after the velox dependency change is merged so that we can refer to the right prod submodule version number. I will wait till then.

"Table writer 4" is for the 4th commit in a series. I once had the first 3 in :).

**Review by mbasmanova — APPROVED** [review]
@gggrace14 Nice cleanup.

**Pull request merged**